Code/Tools/ProjectManager/Source/ProjectGemCatalogScreen.cpp

```diff
@@ -61,51 +62,82 @@ namespace O3DE::ProjectManager
-            const GemInfo::DownloadStatus status = GemModel::GetDownloadStatus(modelIndex);
-            if (GemModel::GetGemOrigin(modelIndex) == GemInfo::Remote &&
-                !(status == GemInfo::Downloaded || status == GemInfo::DownloadSuccessful))
+            QStringList gemPaths, gemNames;
+            for (const QModelIndex& modelIndex : toBeAdded)
             {
-                QMessageBox::critical(
-                    nullptr, "Cannot add gem that isn't downloaded",
-                    tr("Cannot add gem %1 to project because it isn't downloaded yet or failed to download.")
-                        .arg(GemModel::GetDisplayName(modelIndex)));
+                // make sure any remote gems we added were downloaded successfully
+                const GemInfo::DownloadStatus status = GemModel::GetDownloadStatus(modelIndex);
+                if (GemModel::GetGemOrigin(modelIndex) == GemInfo::Remote &&
+                    !(status == GemInfo::Downloaded || status == GemInfo::DownloadSuccessful))
+                {
+                    QMessageBox::critical(
+                        nullptr,
+                        "Cannot add gem that isn't downloaded",
+                        tr("Cannot add gem %1 to project because it isn't downloaded yet or failed to download.")
+                            .arg(GemModel::GetDisplayName(modelIndex)));
+
+                    return ConfiguredGemsResult::Failed;
+                }
+
+                gemPaths.append(GemModel::GetPath(modelIndex));
+                gemNames.append(GemModel::GetName(modelIndex));
+            }
 
-                return ConfiguredGemsResult::Failed;
+            // check compatibility of all gems
+            auto incompatibleResult = pythonBindings->GetIncompatibleProjectGems(gemPaths, gemNames, projectPath);
+            if (incompatibleResult && !incompatibleResult.GetValue().empty())
+            {
+                const auto& incompatibleGems = incompatibleResult.GetValue();
+                QString messageBoxQuestion =
+                    gemNames.length() == 1 ? tr("Do you still want to add this gem?") : tr("Do you still want to add these gems?");
+                QString messageBoxText = QString(tr("%1\n\n%2")).arg(incompatibleGems.join("\n")).arg(messageBoxQuestion);
+                QMessageBox::StandardButton forceAddGems = QMessageBox::warning(this, tr("Gem compatibility issues found"), messageBoxText,
+                    QMessageBox::Yes | QMessageBox::No);
+                if (forceAddGems != QMessageBox::StandardButton::Yes)
+                {
+                    return ConfiguredGemsResult::Cancel;
+                }
             }
 
-            const AZ::Outcome<void, AZStd::string> result = pythonBindings->AddGemToProject(gemPath, projectPath);
-            if (!result.IsSuccess())
+            // we already checked compatibility, so bypass compatibility checks by using 'force'
```

Comment:
Any reason we check for compaibility first before calling `AddGemsToProject`? 
Judging from this comment, it already has compatibility checks. It seems like it would be simpler to just call `AddGemsToProject` without bypassing compatibility checks, unless I am missing an important detail.

Reply:
I initially did use a single function, but chose to break the compatibility check separately too provide better error messaging because our current error handling isn't super great at distinguishing compatibility errors vs everything else.  I also am about to need this function or something similar in an upcoming task to show gem compatibility information in the Gem Catalog.